File: frontend/src/test/factories.ts
import type { FetchUsersResponseDto, UserDto } from '../api/dto';

export function buildUserDto(overrides: Partial<UserDto> = {}): UserDto {
  const id = overrides.id ?? 1;
  return {
    id,
    name: `User ${id}`,
    username: `user${id}`,
    email: `user${id}@example.com`,
    phone: '11987654321',
    ...overrides,
  };
}

export function buildFetchUsersResponseDto(
  overrides: Partial<FetchUsersResponseDto> = {},
): FetchUsersResponseDto {
  return {
    users: [buildUserDto({ id: 1 }), buildUserDto({ id: 2 })],
    failed: [3],
    ...overrides,
  };
}

File: backend/tests/factories.py
from app.domain.user import User


def build_user(user_id: int, phone: str | None = "11987654321") -> User:
    return User(
        id=user_id,
        name=f"User {user_id}",
        username=f"user{user_id}",
        email=f"user{user_id}@example.com",
        phone=phone,
    )
